gcs/conn.go

```diff
@@ -151,7 +157,7 @@ type conn struct {
 func (c *conn) OpenBucket(
 	ctx context.Context,
 	options *OpenBucketOptions) (b Bucket, err error) {
-	b = newBucket(c.client, c.userAgent, options.Name, options.BillingProject)
+	b = newBucket(c.client, c.url, c.userAgent, options.Name, options.BillingProject)
```

Comment:
I am adding this functionality for gcsfuse specifically.  Are there any other users of this library where I should check for a `nil` url for backwards compatibility?

Reply:
Does anything block this pull request?  Using https://github.com/fsouza/fake-gcs-server makes testing gcsfuse a lot nicer.